src/common/io_exerciser/JsonStructures.cc

```diff
@@ -0,0 +1,574 @@
+#include "JsonStructures.h"
+
+#include "common/ceph_json.h"
+#include "OpType.h"
+
+using namespace ceph::io_exerciser::json;
+
+JSONStructure::JSONStructure(std::shared_ptr<ceph::Formatter> formatter) :
+  formatter(formatter)
+{
+
+}
+
+std::string JSONStructure::encode_json()
+{
+  oss.clear();
+
+  dump();
+  formatter->flush(oss);
+  return oss.str();
+}
+
+OSDMapRequest::OSDMapRequest(const std::string& pool_name,
+                             const std::string& object,
+                             const std::string& nspace,
+                             std::shared_ptr<ceph::Formatter> formatter) :
+  JSONStructure(formatter),
+  pool(pool_name),
+  object(object),
+  nspace(nspace)
+{
+
+}
+
+OSDMapRequest::OSDMapRequest(std::shared_ptr<ceph::Formatter> formatter) :
+  JSONStructure(formatter)
+{
+
+}
+
+void OSDMapRequest::decode_json(JSONObj* obj)
+{
+  JSONDecoder::decode_json("prefix", prefix, obj);
+  JSONDecoder::decode_json("pool", pool, obj);
+  JSONDecoder::decode_json("object", object, obj);
+  JSONDecoder::decode_json("nspace", nspace, obj);
+  JSONDecoder::decode_json("format", format, obj);
+}
+
+void OSDMapRequest::dump() const
+{
+  formatter->open_object_section("OSDMapRequest");
```

Comment:
I wonder if the resulting compiled code would be better if all dump()'s were to receive the 'regular pointer to the formatter' as a parameter